src/test/java/com/ksm/wstbackoffice/controller/AddressControllerIT.java

```diff
@@ -0,0 +1,70 @@
+package com.ksm.wstbackoffice.controller;
+
+import com.ksm.wstbackoffice.endpoint.EndPoint;
+import io.restassured.http.ContentType;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.jdbc.SqlGroup;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static io.restassured.RestAssured.given;
+import static io.restassured.RestAssured.when;
+import static org.hamcrest.Matchers.equalTo;
+
+@SqlGroup({
+        @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD, scripts="classpath:sql/addressInsert.sql"),
+        @Sql(executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD, scripts="classpath:sql/addressDrop.sql")
+})
+public class AddressControllerIT extends BaseControllerIT {
+    @Test
+    public void findByIdTest() {
+        given().
+            pathParam("id", 1).
+        when().
+            get(EndPoint.ADDRESSES_ID).
+        then().
+            statusCode(200).
+            body("id", equalTo(1)).
+            body("country.iso3166", equalTo("804"));
+    }
+
+    @Test
+    public void findAllTest() {
+        when().
+            get(EndPoint.ADDRESSES).
+        then().
+            statusCode(200).
+            body("isEmpty()", Matchers.is(false)).
+            body("size()", equalTo(4));
+    }
+
+    @Test
+    public void createTest() {
+        Map<String, Object> country = new HashMap<>();
+        country.put("iso3166", "076");
+        country.put("name", "Brazil");
+
+        Map<String, Object> address = new HashMap<>();
+        address.put("zipcode", "13010");
+        address.put("region", "Brazil region");
+        address.put("district", "Brazil district");
+        address.put("city", "Rio de Janeiro");
+        address.put("street", "Rio Carnival");
+        address.put("houseNumber", "8a");
+        address.put("apartmentNumber", "89");
+        address.put("description", "Brazil address description");
+        address.put("country", country);
+
+        given().
+            contentType(ContentType.JSON).
+            body(address).
+        when().
+            post(EndPoint.ADDRESSES).
+        then().
+            statusCode(201).
+            body("id", equalTo(5));
```

Comment:
It´s difficult to be sure that id is 5. What if somebody will add another createAddress test? It will generate id 5 for it and your test can receive id 6 or any next one.
So in this case it´s enough to check that id is not empty.
What is really important is to check that returned address equals to the prepared for creation.
Probably in this case you would need to extract address from the response and check it equals to the address variable (from the line 50) or something like that.

Reply:
done